contributors/design-proposals/storage/volume-tests.md

```diff
@@ -0,0 +1,154 @@
+
+This should be fixed, we want to run tests in parallel, even with several servers on the same node.
+
+## Proposed changes
+
+* Remove `[Slow]` from `volume_provisioning.go`. On GCE, it tests 3 storage classes in 46 seconds. We have ~7 storage classes on AWS, it could take 2-3 minutes and it's not that slow. There is no `-slow` suite on AWS that would cover them.
+
+* Rework Ceph server image to be able to run multiple times on a node.
+
+* Rework iSCSI server image to be able to run multiple times on a node.
+
+* Add a new test job that will run tests for all volume plugins incl. iSCSI and Ceph. This requires multiple changes covered in the chapter below.
+
+### New job for volume plugin tests.
+As written above, iSCSI, Ceph RBD and CephFS test have `[Feature:Volumes]` tag to be skipped on platforms that don't provide Ceph or iSCSI client utilities and/or kernel modules. No job provides these utilities and the tests don't run.
+
+In order to run these tests, we need:
+* **Run mount utilities in containers instead on the host**. Kubelet already has [MountContainers alpha  feature](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/storage/containerized-mounter-pod.md#implementation-notes) that makes kubelet to run these mount utilities in containers (pods) instead on the host.  From OS point of view, it works the same as CSI, running tests with containerized mount utilities will help us reveal regressions / issues early before CSI drivers for volume plugins are created.
+
+	* **Prepare a container image with mount utilities for NFS, Gluster, iSCSI, Ceph RBD and CephFS**. There is proof-of-concept in [jsafrane/mounter-daemonset repo](https://github.com/jsafrane/mounter-daemonset). It will end up in `test/images/volume-tester/mount`.
+
+	* **Add new option `--deploy-storage-utilities` parameters to `test/e2e.go`**. This will cause E2E test to install a DaemonSet with the aforementioned container on all nodes in`SynchronizedBeforeSuite`. All nodes then can use NFS, Gluster, iSCSI, Ceph RBD and CephFS volumes using pods from the DaemonSet. This option will be off by default.
+
+* **Create a new job `pull-kubernetes-gce-volumes`** that:
+	* Installs Ubuntu cluster with`MountContainers` alpha feature enabled
+	  *	Ubuntu is used to get all necessary kernel modules. COS does not ship them.
+	  *	`MountContainers` alpha feature allows kubelet to run mount utilities in containers instead on the host.
+  * Runs the tests with `e2e.test --deploy-storage-utilities` to deploy the mount utilities for kubelet in containers.
+  * Runs all storage tests with: `--ginkgo.focus=[sig-storage] --ginkgo.skip=[Distruptive]|[Flaky]|[Serial]|[Feature:<all features except Volumes>]`.
+	  * This naturally includes `[Slow]`. See below for experimental run results.
+	  * We want `[Feature:Volumes]` in and  all other `[Feature:.*]` out.
```

Comment:
We may just need to explicitly exclude volume alpha features (like Block)

Reply:
They have different tag, e.g. `[Feature:BlockVolume]`, that get caught by `--ginkgo.skip=...|[Feature:<all features except Volumes>]`.